src/main/java/org/grohe/ondus/api/WebFormLogin.java

```diff
@@ -65,22 +72,45 @@ private String formTargetOf(String page) throws IOException {
 		}
 	}
 
-	private RefreshTokenResponse login(CloseableHttpClient httpclient, String actionUrl) throws IOException {
+	private RefreshTokenResponse login(CloseableHttpClient httpclient, String actionUrl) throws IOException, LoginException {
 		HttpPost post = new HttpPost(actionUrl);
 		post.setHeader("Content-Type", "application/x-www-form-urlencoded");
 		post.setHeader("X-Requested-With", "XMLHttpRequest");
 		post.setHeader("referer", actionUrl);
 		post.setHeader("origin", baseUrl);
-
 		post.setEntity(new StringEntity(buildLoginRequest()));
-
+		
 		try (CloseableHttpResponse response = httpclient.execute(post)) {
+			checkResponse(response);
+
 			return fetchToken(httpclient, fetchLocation(response));
 		}
 	}
 
-	private String fetchLocation(CloseableHttpResponse response) {
-		return response.getHeaders("Location")[0].getValue().replace("ondus://", "https://");
+	private String fetchLocation(CloseableHttpResponse response) throws LoginException, ParseException, IOException {
+		Header[] headers = response.getHeaders("Location");
+		
+		if (headers.length > 0) {
+			return headers[0].getValue().replace("ondus://", "https://");
+		}
+		else {
+			if (checkIsInvalidCredentialsResult(response)) {
+				throw new LoginException("Invalid username/password");
+			}
+			else {
+				throw new LoginException("Unexpected response from grohe webservice");
+			}
+		}
+	}
+
+	private void checkResponse(CloseableHttpResponse response) throws LoginException {
```

Comment:
what about a better bethod name, something like "assertOkResponse", so anyone knows what it does and that it may throw an exception (apart from the throws declaration).